### PR TITLE
Unify Product Elements inserter category to "WooCommerce Product Elements"

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

During some work I realized only some of the Product Elements are placed in the "WooCommerce Product Elements" category while there's a group of blocks to which we refer as Product Elements and that may be confusing to users.

This PR unifies the inserter category for all Product Elements. You can see two types of changes:
- `category: 'woocommerce'` is removed - it's the case for blocks that use [shared config](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/shared/config.tsx#L18) which defines category,
- `category: 'woocommerce'` is replaced with `category: 'woocommerce-product-elements'` - it's the case for blocks that use shared config. I didn't want to spend time making them rely on shared config hence just a replacement.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Product Collection

1. Create a new post
2. Insert Product Collection block and choose some collection (e.g. Featured)
3. Open inserter inside Product Collection block and click "Browse All"

image

4. Scroll to the bottom
5. Expected: There's 7 Product Elements available in "WooCommerce Product Elements" category

Before | After
-- | -- 
image | image

#### Single Product block

1. Create a new post
2. Insert Single Product block and choose some product
3. Open inserter inside Single Product block and click "Browse All"

image

4. Scroll to the bottom
5. Expected: There's 14 Product Elements available in "WooCommerce Product Elements" category. (Note: Image below shows 15, but "Add to Cart" block is experimental and won't display in production build!!!)

Before | After
-- | -- 
image | image

#### Single Product template

1. Go to Editor -> Templates -> Single Product template
2. Open the nserter somewhere on a template and click "Browse All"
3. Scroll to the bottom
4. Expected: There's 7 Product Elements available in "WooCommerce Product Elements" category

Before | After
-- | -- 
image | image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/block.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
 	"icon": "info",
 	"title": "Product Details",
 	"description": "Display a product's description, attributes, and reviews.",
-	"category": "woocommerce",
```
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/block.json:7]: In this case, the block.json isn't extended, so we have to decide if we should keep `woocommerce` or `woocommerce-product-elements`.

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-image-gallery/block.json [hunk 2]
```diff
@@ -4,7 +4,6 @@
 	"title": "Product Image Gallery",
 	"icon": "gallery",
 	"description": "Display a product's images.",
-	"category": "woocommerce",
```
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-image-gallery/block.json:7]: Same here!

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-reviews/block.json [hunk 3]
```diff
@@ -4,7 +4,6 @@
 	"icon": "admin-comments",
 	"title": "Product Reviews",
 	"description": "A block that shows the reviews for a product.",
-	"category": "woocommerce",
```
Comment (gigitux) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-reviews/block.json:7]: same here

**Review by gigitux — CHANGES_REQUESTED** [review]
Thanks for working on this 🙇! I noticed that some blocks don't extend the block.json, and currently, they appear as "uncategorized":

image

We should decide if we use `woocommerce` or `woocommerce-product-elements`. Personally, I would say to use `woocommerce-product-elements`, what do you think?

**Comment by kmanijak** [comment]
>  I noticed that some blocks don't extend the block.json, and currently, they appear as "uncategorized":

@gigitux , apologies for missing that out 🤦 I was checking for "Uncategorized" in Single Product block but missed there are blocks registered for Single Product _template_ only. Thanks for catching this!!!

> We should decide if we use woocommerce or woocommerce-product-elements. Personally, I would say to use woocommerce-product-elements, what do you think?

Definitely. Pushed the change for that. Also, I updated testing steps so it covers three scenarios:
- inside Product Collection block
- inside Single Product block
- in Single Product template

**Review by gigitux — APPROVED** [review]
Thanks for the quick fix and for updating the testing instructions! LGMT! 🚀

**Pull request merged**